e2e_tests/tests/all_providers/mod.rs

```diff
@@ -113,3 +114,31 @@ fn sign_verify_with_provider_discovery() -> Result<()> {
     let key_name = String::from("sign_verify_with_provider_discovery");
     client.generate_rsa_sign_key(key_name)
 }
+
+#[test]
+fn list_keys() {
+    let mut client = TestClient::new();
+
+    let keys = client.list_keys().expect("list_keys failed");
+
+    assert!(keys.is_empty());
```

Comment:
If there are other concurrent tests here, this might fail. I think easiest way out is to just set a different identity for the client used in this test.